ts-client/src/ValorClient.ts

```diff
@@ -0,0 +1,337 @@
+    dataset: string,
+    taskType: TaskType
+  ): Promise<Evaluation> {
+    const response = await this.client.post('/evaluations', {
+      model_names: [model],
+      datum_filter: { dataset_names: [dataset] },
+      parameters: { task_type: taskType }
+    });
+    return response.data[0];
+  }
+
+  /**
+   * Fetches evaluations matching the filters defined by queryParams. This is
+   * private because we define higher-level methods that use this.
+   *
+   * @param queryParams An object containing query parameters to filter evaluations by.
+   *
+   * @returns {Promise<Evaluation[]>}
+   */
+  private async getEvaluations(queryParams: object): Promise<Evaluation[]> {
+    const response = await this.client.get('/evaluations', { params: queryParams });
+    return response.data;
+  }
+
+  /**
+   * Fetches an evaluation by id
+   *
+   * @param id id of the evaluation
+   *
+   * @returns {Promise<Evaluation>}
+   */
+  public async getEvaluationById(id: number): Promise<Evaluation> {
+    const evaluations = await this.getEvaluations({ evaluation_ids: id });
+    return evaluations[0];
+  }
+
+  /**
+   * Fetches all evaluations associated to given models
+   *
+   * @param modelNames names of the models
+   *
+   * @returns {Promise<Evaluation[]>}
+   */
+  public async getEvaluationsByModelNames(modelNames: string[]): Promise<Evaluation[]> {
+    // turn modelNames into a comma-separated string
+    return this.getEvaluations({ models: modelNames.join(',') });
+  }
+
+  /**
+   * Fetches all evaluations associated to given datasets
+   *
+   * @param datasetNames names of the datasets
+   *
+   * @returns {Promise<Evaluation[]>}
+   */
+  public async getEvaluationsByDatasetNames(
+    datasetNames: string[]
+  ): Promise<Evaluation[]> {
+    return this.getEvaluations({ datasets: datasetNames.join(',') });
+  }
+
+  /**
+   * Adds ground truth annotations to a dataset
+   *
+   * @param datasetName name of the dataset
+   * @param datumUid uid of the datum
+   * @param annotations annotations to add
+   *
+   * @returns {Promise<void>}
+   */
+  public async addGroundTruth(
+    datasetName: string,
+    datumUid: string,
+    annotations: object[]
+  ): Promise<void> {
+    await this.client.post('/groundtruths', [
+      { datum: { uid: datumUid, dataset_name: datasetName }, annotations: annotations }
+    ]);
+  }
+
+  /**
+   * Adds predictions from a model
+   *
+   * @param modelName name of the model
+   * @param datasetName name of the dataset
+   * @param datumUid uid of the datum
+   * @param annotations annotations to add
+   *
+   * @returns {Promise<void>}
+   */
+  public async addPredictions(
+    modelName: string,
+    datasetName: string,
+    datumUid: string,
+    annotations: object[]
+  ): Promise<void> {
+    await this.client.post('/predictions', [
+      {
+        model_name: modelName,
+        datum: { uid: datumUid, dataset_name: datasetName },
+        annotations: annotations
+      }
+    ]);
+  }
```

Comment:
better to always return the response IMO so clients have the option to do something with it.

Reply:
updated in [9b1912a](https://github.com/Striveworks/valor/pull/488/commits/9b1912a2347783be4eccbd91571bc53adac1233c)